src/main/java/com/example/bookstore/exception/GlobalExceptionHandler.java

```diff
@@ -25,13 +28,19 @@ public ErrorResponseDto handleValidationException(
                 .map(DefaultMessageSourceResolvable::getDefaultMessage)
                 .toList();
         return generateExceptionDetails(
-            "Input arguments validation failure", HttpStatus.BAD_REQUEST, errors);
+            "Input arguments validation failure", BAD_REQUEST, errors);
+    }
+
+    @ExceptionHandler(RegistrationException.class)
+    public ErrorResponseDto handleRegistrationException(RegistrationException e) {
+        return generateExceptionDetails("Registration error occurred", BAD_REQUEST,
+            List.of(e.getMessage()));
     }
 
     @ExceptionHandler(Exception.class)
     public ErrorResponseDto handleAllExceptions(Exception e) {
         return generateExceptionDetails("An unexpected error occurred",
-            HttpStatus.INTERNAL_SERVER_ERROR, List.of(e.getMessage()));
+            INTERNAL_SERVER_ERROR, List.of(e.getMessage()));
```

Comment:
```suggestion
            INTERNAL_SERVER_ERROR, List.of());
```
It's quite dangerous to return exception messages of exceptions we don't know about (usually programmers' errors on which we return 500 status) because we can show for potential hackers some vulnerabilities of our system.